code/datums/gamemode/role/time_agent.dm

```diff
@@ -0,0 +1,193 @@
+/**
+	Time agent
+
+	You've got 5 minutes of playtime to do something weird
+
+	If you succeed, the stations threat level is escalated
+
+	If you fail, something weird happens.
+
+	Are you a bad enough dude to make sure a corgi, a rubber duck, and a bucket are in the same place at the same time?
+
+**/
+
+/datum/role/time_agent
+	name = "time agent"
+	id = TIMEAGENT
+	required_pref = ROLE_MADNESS
+	logo_state = "time-logo"
+	var/list/objects_to_delete = list()
+
+
+/datum/role/time_agent/Greet(var/greeting,var/custom)
+	if(!greeting)
+		return
+
+	var/icon/logo = icon('icons/logos.dmi', logo_state)
+	switch(greeting)
+		if (GREET_CUSTOM)
+			to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>[custom]</span>")
+		else
+			to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>You are a Time Agent.<br>Specifically you are a scientist by the name of John Beckett, having discovered a method to travel through time, and becoming lost to it. <br>\
```

Comment:
Lore is cool. But when someone pops into existence and they need to get their bearings and worry about people validing them, they need as little wall of text as possible. Could some of this be on the wiki and paraphrased?

Reply:
Wiki's much more freely editable, but yeah I'll drop the loreparagraph